include/nektar_interface/function_projection.hpp

```diff
@@ -11,9 +11,12 @@
 #include <neso_particles.hpp>
 
 #include "basis_reference.hpp"
+#ifdef NESO_USE_NEW_PROJECTION
```

Comment:
I think it would be preferable if this flag enabled the new projection in addition to the existing implementation if we are keeping the existing implementation. Alternatively we replace the existing implementation with the new implementation. 

What I am trying to avoid is a situation where there are two implementations that are selected at compile time as this doubles the number of times we have to compile and run the test suite for each PR.

Reply:
As above I'm nervous about nuking the old versions until it has some more "real world" testing. 
I can do something with an environment variables to make it switchable at runtime so we don't need to mess around with arguments? I will also make the new ones the default.